### PR TITLE
[FEATURE REQUEST] Allow access from document provider preference

### PR DESCRIPTION
**Pull request by JuancaG05**
## Related Issues
App: https://github.com/owncloud/android/issues/3379

- [x] Added changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
_____

## QA

Test plan: https://github.com/owncloud/QA/blob/master/Mobile/Android/Release_2.20/3384-access-document-provider.md

Reports:

### CODE REVIEW

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsSecurityFragment.kt [hunk 1]
```diff
@@ -219,10 +248,11 @@ class SettingsSecurityFragment : PreferenceFragmentCompat() {
         }
     }
 
-    private fun enableBiometricAndLockApplication() {
+    private fun enableBiometricAndLockApplicationAndAccessFromDocumentProvider() {
         prefBiometric?.isEnabled = true
         prefBiometric?.summary = null
```
Comment (fesave) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsSecurityFragment.kt:253]: Use `apply{ ... }`

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsSecurityFragment.kt [hunk 2]
```diff
@@ -231,8 +261,14 @@ class SettingsSecurityFragment : PreferenceFragmentCompat() {
         prefBiometric?.summary = getString(R.string.prefs_biometric_summary)
     }
 
+    private fun disableAccessFromDocumentProvider() {
+        prefAccessDocumentProvider?.isChecked = false
+        prefAccessDocumentProvider?.isEnabled = false
```
Comment (fesave) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsSecurityFragment.kt:266]: Use `apply{ ... }`

**Review by fesave — CHANGES_REQUESTED** [review]

**Review by fesave — APPROVED** [review]
LGTM

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsSecurityFragment.kt [hunk 3]
```diff
@@ -219,20 +248,33 @@ class SettingsSecurityFragment : PreferenceFragmentCompat() {
...
 
     private fun disableBiometric() {
-        prefBiometric?.isChecked = false
-        prefBiometric?.isEnabled = false
-        prefBiometric?.summary = getString(R.string.prefs_biometric_summary)
+        prefBiometric?.apply {
+            isChecked = false
+            isEnabled = false
+            summary = getString(R.string.prefs_biometric_summary)
+        }
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsSecurityFragment.kt:265]: 💯

**Review by abelgardep — APPROVED** [review]
Let's QA this 👍

File: owncloudApp/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -43,6 +43,15 @@
     <string name="prefs_passcode">Passcode lock</string>
     <string name="prefs_biometric">Biometric lock</string>
     <string name="prefs_biometric_summary">Enable passcode or pattern lock to enable this option</string>
+    <string name="prefs_lock_application">Lock application</string>
+    <string name="prefs_lock_application_entries_immediately">Immediately</string>
+    <string name="prefs_lock_application_entries_1minute">After 1 minute</string>
+    <string name="prefs_lock_application_entries_5minutes">After 5 minutes</string>
+    <string name="prefs_lock_application_entries_30minutes">After 30 minutes</string>
+    <string name="prefs_access_from_document_provider">Access from document provider</string>
+    <string name="prefs_access_from_document_provider_summary">Allow access to the files of ownCloud accounts in the app via the Android native file explorer when there is a lock set.</string>
```
Comment (jesmrec) [inline owncloudApp/src/main/res/values/strings.xml:52]: `ownCloud` word should be replaced by the proper parameter appname

Comment (JuancaG05) [reply owncloudApp/src/main/res/values/strings.xml:52]: I rewrote it avoiding the word ownCloud

**Comment by jesmrec** [comment]
An small suggestion about the warning:

`Use this feature under your own responsibility, all the apps will be able to access your account's files via Android native explorer.`

that sounds la bit critical and dreadful, it could scare users too much. I'd compose a sweeter message, making clear what's going on:

`Use this feature with responsibility. Enabling this feature means that your files will be open for those apps that use the Android native explorer, regardless the locking methods you set.`

any other suggestion? @fesave @abelgardep @michaelstingl

**Comment by jesmrec** [comment]
Thanks for the tiny change.

Approved on my side

**Pull request merged**